diff --git a/node_modules/@types/css-tree/index.d.ts b/node_modules/@types/css-tree/index.d.ts
index e5aa5ac..b9c1d73 100644
--- a/node_modules/@types/css-tree/index.d.ts
+++ b/node_modules/@types/css-tree/index.d.ts
@@ -907,6 +907,8 @@ export class Lexer {
     match(syntax: DSNode | string, value: CssNode | string): LexerMatchResult;
 }
 
+export const lexer: Lexer;
+
 export function fork(extension: {
     atrules?: Record<string, string> | undefined;
     properties?: Record<string, string> | undefined;
