diff --git a/node_modules/jest-preset-stylelint/jest-setup.js b/node_modules/jest-preset-stylelint/jest-setup.js
index 80584ae..2b8b636 100644
--- a/node_modules/jest-preset-stylelint/jest-setup.js
+++ b/node_modules/jest-preset-stylelint/jest-setup.js
@@ -1,8 +1,5 @@
 'use strict';
 
-// Mock should be before stylelint required. Even if it's required inside other modules
-jest.mock('stylelint/lib/utils/getOsEol', () => () => '\n');
-
 const getTestRule = require('./getTestRule');
 
 global.testRule = getTestRule();
