diff --git a/node_modules/postcss-selector-parser/postcss-selector-parser.d.ts b/node_modules/postcss-selector-parser/postcss-selector-parser.d.ts
index fbf040e..9163ab1 100644
--- a/node_modules/postcss-selector-parser/postcss-selector-parser.d.ts
+++ b/node_modules/postcss-selector-parser/postcss-selector-parser.d.ts
@@ -237,9 +237,9 @@ declare namespace parser {
         empty(): this;
         insertAfter(oldNode: Child, newNode: Child): this;
         insertBefore(oldNode: Child, newNode: Child): this;
-        each(callback: (node: Child) => boolean | void): boolean | undefined;
+        each(callback: (node: Child, index: number) => boolean | void): boolean | undefined;
         walk(
-            callback: (node: Node) => boolean | void
+            callback: (node: Node, index: number) => boolean | void
         ): boolean | undefined;
         walkAttributes(
             callback: (node: Attribute) => boolean | void
